Lt missing

PratMis opened the pull request:
Closes #4568 

Added a try/except block around describe_launch_template_versions api call. Continues if the launch template is not found and raises in all other cases

PratMis commented:
Might have to put in a test case I guess. If this looks good @kapilt, I can get the tests in! Thanks

kapilt commented:
approach looks fine wrt to api exception, but there's some additional behavior validation needed around the filter itself, ie if attribute equality check on launch template shouldn't return result. also brings up the another question of how do you check for an asg with a non existent launch template version.

PratMis commented:
Cool, I'll try to add those changes in! Thanks

PratMis commented:
Hello @kapilt! I have added some code:
1) Added an exception handler for non existent launch template id version
2) Appending to result list only if ltv has something. This was added wrt your first ask.

kapilt commented:
needs some additional test coverage on the exception handling behavior, if you need an example of injecting clients with failures,  patch or mock is appropriate, there are other resource tests doing the same.

PratMis commented:
Thanks for your input! I'll get the test cases out such that the codecov goes up. Right now looks like its not respecting the test I wrote as per your suggestion.
Also, I looked into the invalid filter. From what I understand, we can return an ASG as invalid if the launch template/launch-template-version it is tied to doesn't exist. But that wouldn't help in this case as it would make the logic redundant.
